Lib/email/_header_value_parser.py

```diff
@@ -1051,7 +1054,7 @@ def get_encoded_word(value):
     try:
         text, charset, lang, defects = _ew.decode('=?' + tok + '?=')
     except ValueError:
-        raise errors.HeaderParseError(
+        raise errors.InvalidEwError(
```

Comment:
Let's move this exception out of `errors.py` and into this module as `_InvalidEwError` (and then it also probably doesn't need to inherit from `HeaderParseError`).  It's just an internal implementation exception, so let's not expose it, even if it's not in `__all__` or documented.

Reply:
Done. I've kept it inheriting from `HeaderParseError`, though, because there are some tests that rely on a `HeaderParseError` to be raised (when an `_InvalidEwError`) and it didn't seem right to have a test that tests to see if an internal implementation exception is raised.

Reply:
I agree the inheriting from `HeaderParseError` makes sense because of the way it is used internally as both an implementation detail and an external error.